mypy/main.py

```diff
@@ -51,6 +51,13 @@ def main(script_path: Optional[str], args: Optional[List[str]] = None) -> None:
         args: Custom command-line arguments.  If not given, sys.argv[1:] will
         be used.
     """
+    # Check for known bad Python versions.
+    vi = tuple(sys.version_info)
+    if vi[:2] < (3, 4):
+        sys.exit("Running mypy with Python 3.3 or lower is not supported; please upgrade")
+    if vi[:3] == (3, 5, 0):
+        sys.exit("Running mypy with Python 3.5.0 is not supported; please upgrade")
```

Comment:
I think it is worth it to specify that upgrading to a later version of `3.5` is possible. It isn't hard for me to imagine someone seeing this and thinking that they must use `3.6.x` or newer to use mypy, which is not the impression we want. Perhaps `... please upgrade to Python 3.5.1 or newer`?